### PR TITLE
feat: Table cell component for jobs

### PR DESCRIPTION
**Pull request by minghay**
## Context
The new jobs UI makes use of custom components for each cell in the table.  This creates a new component for handling the job name and the build status.

## Objective
Create a glimmer component for the jobs.  Screenshot will be captured in the linked PR

## References
#1171 for screenshot 
https://github.com/screwdriver-cd/screwdriver/issues/3200

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/pipeline/jobs/table/cell/job/util.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * @returns {string}
+ */
+export default function getDisplayName(job) {
+  const jobName = job.name;
+  const { annotations } = job?.permutations?.[0] || {};
+
+  if (annotations) {
+    return annotations['screwdriver.cd/displayName']
+      ? annotations['screwdriver.cd/displayName']
+      : jobName;
```
Comment (tkyi) [inline app/components/pipeline/jobs/table/cell/job/util.js:13]: does this work as well?
```suggestion
    return annotations['screwdriver.cd/displayName'] || jobName;
```

File: app/components/pipeline/jobs/table/cell/job/component.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        this.latestBuild = builds[builds.length - 1];
+        this.statusIcon = statusIcon(this.latestBuild.status);
+      }
+    });
+  }
+
+  willDestroy() {
+    super.willDestroy();
+
+    this.args.record.onDestroy(this.args.record.job);
```
Comment (VonnyJap) [inline app/components/pipeline/jobs/table/cell/job/component.js:27]:
```suggestion
    // Ensure the job exists before attempting to destroy
    if (this.args.record.job) {
      this.args.record.onDestroy(this.args.record.job);
    }
```

File: app/components/pipeline/jobs/table/cell/job/component.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    super(...arguments);
+
+    const { job } = this.args.record;
+
+    this.args.record.onCreate(job, builds => {
+      if (builds.length > 0) {
+        this.latestBuild = builds[builds.length - 1];
+        this.statusIcon = statusIcon(this.latestBuild.status);
+      }
+    });
```
Comment (VonnyJap) [inline app/components/pipeline/jobs/table/cell/job/component.js:21]:
```suggestion
    // Ensure the job exists before setting up the listener
    if (job) {
      this.args.record.onCreate(job, builds => {
        if (builds.length > 0) {
          this.latestBuild = builds[builds.length - 1];
          this.statusIcon = statusIcon(this.latestBuild.status);
        }
      });
    }
```

File: app/components/pipeline/jobs/table/cell/job/template.hbs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        @route="pipeline.build"
+        @models={{array @record.job.pipelineId this.latestBuild.id}}
+      >
+        <FaIcon
+          @icon={{this.statusIcon.name}}
+          @fixedWidth="true"
+          @prefix={{this.statusIcon.prefix}}
+          @spin={{this.statusIcon.spin}}
+          @flip={{this.statusIcon.flip}}
+          class={{this.latestBuild.status}}
```
Comment (VonnyJap) [inline app/components/pipeline/jobs/table/cell/job/template.hbs:14]: do we need to add `alt` attribute?

File: app/components/pipeline/jobs/table/cell/job/component.js (hunk 2 above)
Comment (minghay) [reply app/components/pipeline/jobs/table/cell/job/component.js:27]: This isn't needed as the job is always passed into the component.

File: app/components/pipeline/jobs/table/cell/job/component.js (hunk 3 above)
Comment (minghay) [reply app/components/pipeline/jobs/table/cell/job/component.js:21]: This check is also not needed as the job is always passed into the component.

File: app/components/pipeline/jobs/table/cell/job/template.hbs (hunk 4 above)
Comment (minghay) [reply app/components/pipeline/jobs/table/cell/job/template.hbs:14]: I haven't thought about how best to handle the accessibility portion for the new UI.  Ideally, yes, but there are a number of outstanding accessibility improvements that should be made.  It's probably best to save this change as part of an accessibility improvement issue.

**Review by VonnyJap — APPROVED** [review]

**Pull request merged**